libraries/AP_TECS/AP_TECS.cpp

```diff
@@ -358,36 +353,15 @@ void AP_TECS::_update_speed(float load_factor)
         _TASmin = _TAS_dem;
     }
 
-    // Reset states of time since last update is too large
-    if (DT > 1.0f) {
-        _TAS_state = (_EAS * EAS2TAS);
-        _integDTAS_state = 0.0f;
-        DT            = 0.1f; // when first starting TECS, use a
-        // small time constant
-    }
-
     // Get airspeed or default to halfway between min and max if
     // airspeed is not being used and set speed rate to zero
     bool use_airspeed = _use_synthetic_airspeed_once || _use_synthetic_airspeed.get() || _ahrs.airspeed_sensor_enabled();
-    if (!use_airspeed || !_ahrs.airspeed_estimate(&_EAS)) {
+    if (!use_airspeed || !_ahrs.EAS_estimate(&_EAS)) {
         // If no airspeed available use average of min and max
         _EAS = 0.5f * (aparm.airspeed_min.get() + (float)aparm.airspeed_max.get());
     }
 
-    // Implement a second order complementary filter to obtain a
-    // smoothed airspeed estimate
-    // airspeed estimate is held in _TAS_state
-    float aspdErr = (_EAS * EAS2TAS) - _TAS_state;
-    float integDTAS_input = aspdErr * _spdCompFiltOmega * _spdCompFiltOmega;
-    // Prevent state from winding up
-    if (_TAS_state < 3.1f) {
-        integDTAS_input = MAX(integDTAS_input , 0.0f);
-    }
-    _integDTAS_state = _integDTAS_state + integDTAS_input * DT;
-    float TAS_input = _integDTAS_state + _vel_dot + aspdErr * _spdCompFiltOmega * 1.4142f;
-    _TAS_state = _TAS_state + TAS_input * DT;
-    // limit the airspeed to a minimum of 3 m/s
-    _TAS_state = MAX(_TAS_state, 3.0f);
+    _TAS_state = _EAS * EAS2TAS;
```

Comment:
This is bad. The previous max was preventing us from dividing by 0, or a very small number. This is important as it can have a very large impact on the calculation of `velRateMax` and `velRateMin` in `AP:TECS::_update_speed_demand`.

Reply:
I think _EAS is constrained to a min of 3 m/sec... will check

Reply:
Ah it is, but it needs to be commented up at the header level that it is guaranteed to always be at least 3. This is really going to bite when someone decides to rearrange it in the future :)

Reply:
moved the 3 m/sec constraint out of AP_Airspeed and put it here